### ISSUE TITLE
aws_rds_cluster_parameter_group does not generate a unique name

### ISSUE THREAD

**Issue by tete17** [terraform-providers/terraform-provider-aws#7264]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```bash
~$ terraform -v
Terraform v0.11.11
```

### Affected Resource(s)

* aws_rds_cluster_parameter_group

### Terraform Configuration Files

```hcl
resource "aws_rds_cluster_parameter_group" "parameters" {
  family = "aurora5.6"

  parameter {
    apply_method = "pending-reboot"
    name  = "innodb_file_per_table"
    value = "1"
  }
}
```

### Debug Output

```bash
Error: Error applying plan:

1 error(s) occurred:

* aws_rds_cluster_parameter_group.parameters: 1 error(s) occurred:

* aws_rds_cluster_parameter_group.parameters: Error modifying DB Cluster Parameter Group: InvalidParameterValue: The parameter DBClusterParameterGroupName must be provided and must not be blank.
        status code: 400, request id: bbd822ba-95c3-4c65-...

```

### Expected Behavior

Create The resource

### Actual Behavior

Fails to create it

### Steps to Reproduce

1. `terraform apply`

### References

Maybe related to #1739

**Comment by bflad** [comment]
The fix for the `aws_rds_cluster_parameter_group` resource not correctly applying during resource creation with a generated name or with the `name_prefix` argument while also including `parameter` configurations has been merged and will release with version 1.57.0 of the Terraform AWS provider, likely later today. 👍

**Comment by tete17** [comment]
Amazing how quick it was guys. Thanks :smile:

**Comment by bflad** [comment]
This has been released in [version 1.57.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1570-january-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
